Integrations/src/main/java/io/deephaven/integrations/learn/IndexSet.java

```diff
@@ -0,0 +1,79 @@
+package io.deephaven.integrations.learn;
+
+import java.util.NoSuchElementException;
+import java.util.PrimitiveIterator;
+
+/**
+ * Stores indices from a table and provides an iterator over those indices.
+ */
+public class IndexSet implements Iterable<Long> {
```

Comment:
In my Python gather function, the index set `idx` is one of the arguments, and then I get the iterator in Python using `idx.iterator()`. I suspect that means this needs to be public, but since the Python gather function is actually called from within Java, maybe not. However, I think I had it package private until I couldn't get the gather to work without making it public.